ocaml/libs/tracing/tracing.ml

```diff
@@ -567,11 +567,14 @@ let set ?enabled ?attributes ?endpoints ~uuid () =
     List.for_all
       (fun provider -> not provider.TracerProvider.enabled)
       (get_tracer_providers ())
-  then
+  then (
+    set_observe false ;
     Xapi_stdext_threads.Threadext.Mutex.execute Spans.lock (fun () ->
         Hashtbl.clear Spans.spans ;
```

Comment:
maybe not related to this PR, but do we need to flush the spans before clearing them?

Reply:
I think the finished spans will still be exported by the exporter after some time. This clears the ongoing spans when all tracer providers are disabled.

Reply:
Ok, but line 574 is also clearing the `finished_spans` is that fine?